facade-charm/tests/integration/test_charm.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+# Copyright 2024 pietro
+# See LICENSE file for licensing details.
+
+import asyncio
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
+    """Build the charm-under-test and deploy it together with related charms.
+
+    Assert on the unit status before any relations/configurations take place.
+    """
+    # Build and deploy charm from local source folder
+    charm = await ops_test.build_charm(".")
+    resources = {"httpbin-image": METADATA["resources"]["httpbin-image"]["upstream-source"]}
+
+    # Deploy the charm and wait for active/idle status
+    await asyncio.gather(
+        ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME),
+        ops_test.model.wait_for_idle(
+            apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=1000
+        ),
+    )
```

Comment:
I've seen this before, and I still wonder if `gather` is warranted here.

Does it make sense for an app to be idle before it is deployed?

Would the below be equivalent, with the extra benefit of being obvious enough making the comment redundant?

```suggestion
    await ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME)
    await ops_test.model.wait_for_idle(
            apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=1000
        )
```

Reply:
Oh wait... is the whole point to be able to terminate if the unit gets blocked while it is being deployed?

In that case the code makes sense, though I wish the way it was written made it more obvious!

Reply:
this is all the built-in `charmcraft init` profile. I agree with the thoughts but they need to be addressed there, not here :)